src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -288,7 +288,7 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
 
       const returnFiber = workInProgress.return;
 
-      if (returnFiber) {
+      if (returnFiber && !next) {
```

Comment:
Because the first time the coroutine completes, it adds itself to the effect list. Then it does its child, and completes again.

Not sure what happens if there are no children in the first phase. It probably breaks anyway.